release-notes/opensearch-knn.release-notes-2.4.0.0.md

```diff
@@ -0,0 +1,32 @@
+## Version 2.4.0.0 Release Notes
+
+Compatible with OpenSearch 2.4.0
+
+### Enhancements
+* Merge efficient filtering from feature branch ([#589](https://github.com/opensearch-project/k-NN/pull/589)) [2.x]
```

Comment:
I think we will usually list all the PRs that we merged to main branch unless it is merged only to 2.x branch

Reply:
Oops. I also need to remove [2.x] part. Forgot that.